version: "3.9"

services:
  auth-db:
    container_name: auth-db
    image: postgres:latest
    env_file:
      - gateway/.env.postgres
    volumes:
      - pgdata:/var/lib/postgresql/data/
    ports:
      - "5432:5432"
    restart: on-failure
    networks:
      - private_network

  auth-cache:
    container_name: auth-cache
    image: redis:latest
    ports:
      - "6379:6379"
    volumes:
      - redisdata:/data
      - redisconfig:/etc/redis
    restart: on-failure
    networks:
      - private_network

  auth-gateway:
    container_name: auth-gateway
    build:
      context: ./gateway
    working_dir: /gateway
    command: uvicorn server:main --host 0.0.0.0 --port 8000 --reload
    environment:
      - OAUTHLIB_INSECURE_TRANSPORT=1
    env_file:
      - gateway/.env
    volumes:
      - ./gateway:/gateway
    ports:
      - "8000:8000"
    restart: on-failure
    networks:
      - public_network
      - private_network
    depends_on:
      - auth-db
      - auth-cache

  shop-api:
    container_name: shop-api
    build:
      context: ./shop
    working_dir: /shop
    command: npm start
    ports:
      - "5000:5000"
    restart: on-failure
    volumes:
      - ./shop:/shop
    networks:
      - private_network

volumes:
  pgdata:
  redisdata:
  redisconfig:

networks:
  private_network:
  public_network:
